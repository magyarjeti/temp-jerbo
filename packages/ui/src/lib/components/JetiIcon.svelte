<script lang="ts">
	import { concatClasses } from '@packages/toolbox/concatClasses';
	import type { Snippet } from 'svelte';

	type Props = {
		children?: Snippet;
		class?: string | null;
	};
	let { children, ...props }: Props = $props();
</script>

<span class={concatClasses(props.class, 'jeti-icon')}>{@render children?.()}</span>

<style>
	@font-face {
		src: url('https://cdn.444r.cloud/assets/SkanIcons-Regular-EYGT6Q66.de61f1566e9fde96cd5b.ttf');
		font-family: 'JetiIcons';
	}

	.jeti-icon {
		display: inline-block;
		text-transform: none;
		text-rendering: optimizeLegibility;
		line-height: 1;
		font-family: 'JetiIcons';
		font-weight: normal;
		font-style: normal;
		font-feature-settings: liga;
		font-size: calc(var(--icon-size, 24) * 1px);
		letter-spacing: normal;
		word-wrap: normal;
		white-space: nowrap;
		direction: ltr;
		user-select: none;
	}
</style>
