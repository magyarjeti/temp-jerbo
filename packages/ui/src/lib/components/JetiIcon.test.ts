import { render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import JetiIcon from './JetiIcon.svelte';

test('JetiIcon snapshot', () => {
	const { container } = render(JetiIcon);
	expect(container).toMatchSnapshot();
});
