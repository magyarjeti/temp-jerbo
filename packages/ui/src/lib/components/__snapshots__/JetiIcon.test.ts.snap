// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JetiIcon snapshot 1`] = `
<div>
  <span
    class="jeti-icon svelte-1pzkbd4"
  />
</div>
`;
